### PR TITLE
Update training_args.py - addition of self.distributed_state when using XPU

### PR DESCRIPTION
**Pull request by Serizao**
# addition of `self.distributed_state` when using XPU

# Fixe

In the base code self.distributed_state does not appear to be defined, which causes the script to crash when used on lines 1813 and 1814 in my case.

I therefore propose an update with a definition of this variable when using an XPU

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by amyeroberts** [comment]
cc @muellerzr @pacman100

**Review by lawchingman — APPROVED** [review]

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1791,6 +1791,7 @@ def _setup_devices(self) -> "torch.device":
             self._n_gpu = 1
             torch.cuda.set_device(device)
         elif is_torch_xpu_available() and "ACCELERATE_USE_XPU" not in os.environ:
+            self.distributed_state = PartialState(timeout=timedelta(seconds=self.ddp_timeout))
             os.environ["ACCELERATE_USE_XPU"] = "true"
```
Comment (muellerzr) [inline src/transformers/training_args.py:1795]:
```suggestion
            os.environ["ACCELERATE_USE_XPU"] = "true"
            self.distributed_state = PartialState(timeout=timedelta(seconds=self.ddp_timeout))
```

**Review by muellerzr — COMMENTED** [review]
Nice catch! We should also change the `os.environ` to be in front so `PartialState` picks it up

**Review by muellerzr — APPROVED** [review]
Thanks! LG2M here

**Review by amyeroberts — APPROVED** [review]
Thanks for adding! 

Before merging - could you: 
* Update the PR title to something more descriptive
* Give a bit more context about the error? 

The addition makes sense and is in line with the other logic. However, `self.distributed_state` is defined on L1782 and the referenced line in the PR description L1813-1814 doesn't seem to be related to this issue. This is mainly for documentation if people come back to this PR.

**Pull request merged**